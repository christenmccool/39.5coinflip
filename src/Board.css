.Board {
  text-align: center;
}

.Board-btn {
  display: block;
  margin: 20px auto;
  padding: 5px 10px;
  border-radius: 8px;
}