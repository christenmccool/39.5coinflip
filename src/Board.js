import React, { useState } from "react";
import './Board.css';
import Coin from './Coin.js';

const Board = () => {

  function flipCoin() {
    if (Math.random() > 0.5) {
      return "heads";
    } else {
      return "tails";
    }
  }

  const [side, setSide] = useState(null);
  const [total, setTotal] = useState(0);
  const [headCt, setHeadCt] = useState(0);
  const [tailCt, setTailCt] = useState(0);

  const handleClick = () => {
    const newFlip = flipCoin();
    setSide(newFlip);

    setTotal(total + 1);
    // if (side === 'heads') {
    if (newFlip === 'heads') {
      setHeadCt(headCt + 1);
    } else {
      setTailCt(tailCt + 1);
    }
  }

  const CoinEle = side ? <Coin side={side} /> : null;

  return (
    <div className="Board">
      <h1 className="Board-title">Let's flip a coin!</h1>

      {CoinEle}

      <button className="Board-btn" onClick={handleClick}>Flip a coin</button>

      <h3 className="Board-count">
        Out of {total} flips, there have been {headCt} heads and {tailCt} tails.
      </h3>
    </div>
  )
}

export default Board;