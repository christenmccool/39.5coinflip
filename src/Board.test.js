import { render, fireEvent } from '@testing-library/react';
import Board from './Board';

beforeEach(function() {
  jest
    .spyOn(Math, "random")
    .mockReturnValueOnce(0.25)
    .mockReturnValueOnce(0.75);
});

afterEach(function() {
  Math.random.mockRestore();
});

test("renders without crashing", () => {
  render(<Board />)
})

test("matches snapshot", () => {
  const {asFragment} = render(<Board />);
  expect(asFragment()).toMatchSnapshot();
})

test("no coin image on load", () => {
  const { queryByAltText, debug } = render(<Board />);

  expect(queryByAltText('tails')).not.toBeInTheDocument();
  expect(queryByAltText('heads')).not.toBeInTheDocument();
  // debug();
})

test("coin image on first button click", () => {
  const { getByText, queryByAltText, debug } = render(<Board />);

  const btn = getByText("Flip a coin");

  fireEvent.click(btn);
  expect(queryByAltText('tails')).toBeInTheDocument();
  // debug();
})

test("coin image change on second button click", () => {
  const { getByText, queryByAltText, debug } = render(<Board />);

  const btn = getByText("Flip a coin");

  fireEvent.click(btn);
  fireEvent.click(btn);

  expect(queryByAltText('tails')).not.toBeInTheDocument();
  expect(queryByAltText('heads')).toBeInTheDocument();

  // debug();
})

test("initial counts are zero", () => {
  const { getByText, debug } = render(<Board />);

  expect(getByText('Out of 0 flips, there have been 0 heads and 0 tails.')).toBeInTheDocument();
})

test("count updates on first click", () => {
  const { getByText, debug } = render(<Board />);

  const btn = getByText("Flip a coin");

  fireEvent.click(btn);

  expect(getByText('Out of 1 flips, there have been 0 heads and 1 tails.')).toBeInTheDocument();
})

test("count updates on second click", () => {
  const { getByText, debug } = render(<Board />);

  const btn = getByText("Flip a coin");

  fireEvent.click(btn);
  fireEvent.click(btn);

  expect(getByText('Out of 2 flips, there have been 1 heads and 1 tails.')).toBeInTheDocument();
})