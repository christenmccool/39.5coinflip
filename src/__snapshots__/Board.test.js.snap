// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Board"
  >
    <h1
      class="Board-title"
    >
      Let's flip a coin!
    </h1>
    <button
      class="Board-btn"
    >
      Flip a coin
    </button>
    <h3
      class="Board-count"
    >
      Out of 0 flips, there have been 0 heads and 0 tails.
    </h3>
  </div>
</DocumentFragment>
`;
